code/modules/urist/datums/xenoarch_effects/poltergeist.dm

```diff
@@ -0,0 +1,42 @@
+/datum/artifact_effect/poltergeist
+	name = "poltergeist"
+	effect_type = EFFECT_ENERGY
+
+
+/datum/artifact_effect/poltergeist/DoEffectTouch(mob/living/user)
+	throw_at_mob(user, rand(10, 30))
+
+
+/datum/artifact_effect/poltergeist/DoEffectAura()
+	var/mob/living/target = null
+	for (var/mob/living/L in oview(get_turf(holder), effectrange))
+		if (L.stat || !L.mind)
+			continue
+		if (target && get_dist(get_turf(holder), L) > get_dist(get_turf(holder), target))
+			continue
+		target = L
+	if (target)
+		throw_at_mob(target, rand(15, 30))
+
+
+/datum/artifact_effect/poltergeist/DoEffectPulse()
+	var/mob/living/target = null
+	for (var/mob/living/L in oview(get_turf(holder), effectrange))
+		if (L.stat || !L.mind)
+			continue
+		if (target && get_dist(get_turf(holder), L) > get_dist(get_turf(holder), target))
+			continue
+		target = L
+	if (target)
+		throw_at_mob(target, chargelevelmax)
+
+
+/datum/artifact_effect/poltergeist/proc/throw_at_mob(mob/living/target, damage = 20)
+	var/list/valid_targets = list()
+	for (var/obj/O in oview(world.view, target))
+		if (!O.anchored && isturf(O.loc))
+			valid_targets |= O
+	if (length(valid_targets))
+		var/obj/obj_to_throw = pick(valid_targets)
```

Comment:
Do we perhaps want to treat targets, e.g. closer or further, preferentially? That would be fairly easy to do since we've already got a PriorityQueue type defined.